### ISSUE TITLE
How to obtain the bioBert embeddings

### ISSUE THREAD

**Issue by chinmay5** [nyuad-cai/CXR-ML-GZSL#3]
Is it possible to know the process of obtaining the BioBERT embeddings? I tried to obtain the values on my own but somehow I am still struggling with obtaining the correct embedding tensor.

**Comment by nasir6** [comment]
we used transformers library https://github.com/huggingface/transformers with 'dmis-lab/biobert-large-cased-v1.1' pre-trained weights and the embeddings were extracted through "last_hidden_state" of the output.

**Comment by alee355** [comment]
I'm also having a trouble of obtaining the BioBERT embedding. How did you guys acquire the penultimate layer of BioBERT for all trainable layers? can you also share the script that you used to acquire the BioBert embedding?

**Comment by chinmay5** [comment]
This is what I ended up using. The results were quite close to the embeddings in the original file

```

import torch
from transformers import (
    AutoConfig,
    AutoTokenizer,
    AutoModel,
    HfArgumentParser,
    set_seed,
    pipeline
)

class BERTEmbedModel:
    def __init__(self):
        model_name_or_path = 'dmis-lab/biobert-large-cased-v1.1'

        config = AutoConfig.from_pretrained(model_name_or_path)
        tokenizer = AutoTokenizer.from_pretrained(model_name_or_path,
            use_fast=False,
        )
        model = AutoModel.from_pretrained(
            model_name_or_path,
            from_tf=bool(".ckpt" in model_name_or_path),
            config=config,
            cache_dir=None,
        )
        self.nlp = pipeline(task="feature-extraction", model=model, tokenizer=tokenizer)

    def get_embeds(self, text):
        output = self.nlp(text)
        wts = []
        for idx in range(len(output[0])):
            wts.append(torch.tensor(output[0][idx]))
        wts = torch.stack(wts).mean(axis=0)
        return wts

if __name__ == '__main__':
    model = BERTEmbedModel()
    print(model.get_embeds("What if a long text is sent").shape)

```

Hope this helps.

**Comment by alee355** [comment]
so the argument text ("What if a long text is sent") would be one of the trainable class right? (e.g. Infiltration, edema)

**Comment by chinmay5** [comment]
@alee355 exactly!!!

**Comment by alee355** [comment]
Thanks a lot!! I really appreciate it.

**Comment by eplatas** [comment]
Hi [chinmay5](https://github.com/chinmay5), Thank you very much for the code and for the explanation, 

I have a question: if I want to visualize the matrix of vectors embeds, I would have to print wts

 def get_embeds(self, text):
        output = self.nlp (text)
        print(text)
        wts = []
        for idx in range(len(output[0])):
            wts.append(torch.tensor(output[0][idx]))
        wts = torch.stack(wts).mean(axis=0)
        **print(wts)**
        return wts

Cheers
